### PR TITLE
Fixing inconsistencies when importing/exporting APIs/API Products/Applications with Throttling Policies using the API Controller

### PR DESCRIPTION
**Pull request by wasuradananjith**
## Purpose
Fixes: https://github.com/wso2/product-apim-tooling/issues/329

## Goals
Eliminate the inconsistencies when adding throttling policies when importing an API/API Product/Application. 

## Approach
The changes have been done related to the three (3) main throttling policies. 
1. **Advanced Throttling Policies**
    1. Modifications related to **API/Product Level Throttling Policies**
        - When we try to import an API/API Product with an Advanced Throttling Policy that has been assigned to the  API/API Product, which is not currently available in the API Manager, an error was displayed earlier. But with this PR, now you will receive an error and the import will be failed.
    2. Modifications related to **Resource Level Throttling Policies**
        - No changes needed to be done.
2. **Application-level Throttling Tiers/Policies**
    - When we try to import an Application, if the Application-level Throttling Tier/Policy is not available in the API Manager instance, the application was imported to that instance, which was wrong. But with this PR, now you will receive an error if you try to do that and the import will be failed.
3. **Subscription-level Throttling Tiers/Policies**
    - When we export an API/API Product with a Subscription-level Throttling Tier/Policy using the API Controller, the throttling policy **details** will be included in the api.yaml file inside the exported directory as an array. But with this PR, the throttling policy **names** will be exported as an array instead of all the **details** of tiers.
    - When we try to import an API/API Product, if any of the Subscription-level Throttling Tiers/Policies were not available in the API Manager, the API Manager ignored it and imported the API/API Product with the existing Subscription-level Throttling Tiers/Policies. But with this PR, now you will receive an error if you are trying to import an API/API Product with the Subscription-level Throttling Tiers/Policies if any of those are not available in the API Manager instance and the import will be failed.

Refer [1] to understand more about the approach.

## User stories
Refer the points in the approach to understand the use cases.

## Test environment
- JDK 1.8.0_251
- Ubuntu 20.04.1 LTS

 [1] http://wso2-oxygen-tank.10903.n7.nabble.com/APIM-Support-to-Import-Export-Throttling-Policies-Using-the-API-Controller-td162176.html

### CODE REVIEW

**Review by uvindra — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIAndAPIProductCommonUtil.java [hunk 1]
```diff
@@ -639,4 +647,74 @@ public static void addClientCertificates(String pathToArchive, APIProvider apiPr
             throw new APIImportExportException(errorMessage, e);
         }
     }
+
+    /**
+     * Get the subscription level policy names of an API/API Product
+     *
+     * @param apiTypeWrapper API or API Product to be exported
+     * @throws APIImportExportException
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIAndAPIProductCommonUtil.java:655]: No exception is thrown in this method. Please correct the java docs.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIAndAPIProductCommonUtil.java [hunk 2]
```diff
@@ -639,4 +647,74 @@ public static void addClientCertificates(String pathToArchive, APIProvider apiPr
...
+        }
+        return tiers;
+    }
+
+    /**
+     * Set the subscription level tiers of an API/API Product by validating with the tiers available in the environment
+     *
+     * @param apiJsonContent JSON content of API or API Product to be imported
+     * @param apiProvider    API Provider
+     * @throws APIImportExportException
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIAndAPIProductCommonUtil.java:676]: APIManagementException is thrown, replace the APIImportExportException with it in the java docs.

Comment (wasuradananjith) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIAndAPIProductCommonUtil.java:676]: Fixed via https://github.com/wso2/carbon-apimgt/commit/eb9b7181c0f6cb5b9624602d91d28a7479bf551b

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIAndAPIProductCommonUtil.java (hunk 1 above)
Comment (wasuradananjith) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIAndAPIProductCommonUtil.java:655]: Fixed via https://github.com/wso2/carbon-apimgt/commit/eb9b7181c0f6cb5b9624602d91d28a7479bf551b

**Pull request merged**